tests/publiccloud/ipa.pm

```diff
@@ -0,0 +1,162 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Use IPA framework to test public cloud SUSE images
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>
+
+use base "consoletest";
+use strict;
+use testapi;
+use utils;
+use File::Basename;
+use serial_terminal 'select_virtio_console';
+
+
+sub find_secgroup {
+    my ($name) = @_;
+
+    my $out = script_output("aws ec2 describe-security-groups --group-names '$name'", 30, proceed_on_failure => 1);
+    if ($out =~ /"GroupId":\s+"([^"]+)"/) {
+        return $1;
+    }
+    return;
+}
+
+sub save_logs {
+    my $output = script_output("find ipa_results -type f");
+    for my $file (split(/\n/, $output)) {
+        if ($file =~ m'ipa_results/ec2/ami-[0-9a-z]+/i-[0-9a-z]+/[0-9]{14}\.(log|results)') {
+            upload_logs($file);
+        }
+    }
+}
+
+sub run {
+
+    select_virtio_console();
+
+    die "Public cloud provider isn't supported" unless check_var('PUBLIC_CLOUD_PROVIDER', "EC2");
+
+    zypper_call('ar ' . get_required_var('PUBLIC_CLOUD_TOOLS_REPO'));
+    zypper_call('--gpg-auto-import-keys in python3-ipa python3-ipa-tests');
+
+    # WAR install awscli from pip instead of using the package bcs#1095041
+    zypper_call('in gcc python3-pip');
+    if (check_var('DISTRI', 'opensuse')) {
+        zypper_call('in python3-devel');
+        assert_script_run("pip3 install pycrypto");
+    }
+    assert_script_run("pip3 install awscli");
+    assert_script_run("pip3 install keyring");
+
+    assert_script_run("export AWS_ACCESS_KEY_ID=" . get_required_var('PUBLIC_CLOUD_KEY_ID'));
+    assert_script_run("export AWS_SECRET_ACCESS_KEY=" . get_required_var('PUBLIC_CLOUD_KEY_SECRET'));
+    assert_script_run("export AWS_DEFAULT_REGION='eu-central-1'");
+
+    # Create SSH key
+    assert_script_run("aws ec2 delete-key-pair --key-name QA_SSH_KEY");
+    assert_script_run("aws ec2 create-key-pair --key-name QA_SSH_KEY --query 'KeyMaterial' --output text > QA_SSH_KEY.pem");
+
+    # Create/find security group
+    my $secgroup_id = find_secgroup("qa_secgroup");
+    if (!defined($secgroup_id)) {
+        assert_script_run("aws ec2 create-security-group --group-name qa_secgroup --description 'SSH_OPEN'");
```

Comment:
I would say that is code is part of find_secgroup. What do you think about putting **everything** related to a search of secgroup inside find_secgroup function?